### PR TITLE
[magnum] Fix register Magnum domain

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

File: chef/cookbooks/magnum/recipes/setup.rb [hunk 1]
```diff
@@ -43,31 +43,47 @@
...
+            --description "Owns users and projects created by magnum" \
+            #{trustee_domain_name})
+        MAGNUM_DOMAIN_ID=$id
+    fi
+
+    [ -n "$MAGNUM_DOMAIN_ID" ] || exit 1
 
     id=
-    eval $(openstack #{insecure} user create \
+    eval $(openstack #{insecure} \
```
Comment (toabctl) [inline chef/cookbooks/magnum/recipes/setup.rb]: Can't we use ```add_user``` from ```keystone_register``` ?

File: chef/cookbooks/magnum/recipes/setup.rb [hunk 2]
```diff
@@ -43,31 +43,47 @@
...
+        domain show \
+        -f shell --variable id \
+        #{trustee_domain_name})
+
+    MAGNUM_DOMAIN_ID=$id
+
+    if [ -z "$MAGNUM_DOMAIN_ID" ]; then
+        id=
+        eval $(openstack #{insecure} \
+            domain create \
```
Comment (toabctl) [inline chef/cookbooks/magnum/recipes/setup.rb]: What about adding a ```:add_domain``` action to ```keystone_register``` and use that?

File: chef/cookbooks/magnum/recipes/setup.rb [hunk 3]
```diff
@@ -43,31 +43,47 @@
...
         --password #{trustee_domain_admin_password} \
         --description "Manages users and projects created by magnum" \
-        #{trustee_domain_admin})
-    ret=
-    eval $(openstack #{insecure} role add \
-        --user $TRUSTEE_DOMAIN_ADMIN_ID \
-        --domain $TRUSTEE_DOMAIN_ID admin)
+        #{trustee_domain_admin}
+
+    openstack #{insecure} \
```
Comment (toabctl) [inline chef/cookbooks/magnum/recipes/setup.rb]: What about adding a ```:add_role``` action to ```keystone_register``` and use that?

Comment (dirkmueller) [inline chef/cookbooks/magnum/recipes/setup.rb]: afaik keystone_register doesn't use v3 so you can't do that

File: chef/cookbooks/magnum/recipes/setup.rb (hunk 1 above)
Comment (mjura) [inline chef/cookbooks/magnum/recipes/setup.rb]: @toabctl: I've checked `keystone_register` resource and the reason, why we have to do this in this way is, that we are creating user with `trustee_domain` . Adding domains is not supported by this resource and better would to keep this in one logical block like for Heat (in Hear we have similar solution)

File: chef/cookbooks/magnum/recipes/setup.rb (hunk 2 above)
Comment (mjura) [inline chef/cookbooks/magnum/recipes/setup.rb]: `:add_domain` doesn't exist for `keystone_register`

File: chef/cookbooks/magnum/recipes/setup.rb [hunk 4]
```diff
@@ -43,31 +43,47 @@
...
   code <<-EOF
+
+    # Find domain ID
     id=
-    eval $(openstack #{insecure} domain create \
-        #{trustee_domain_name} \
-        --description 'Owns users and projects created by magnum' \
-        --or-show -f value -c id)
-    TRUSTEE_DOMAIN_ID=$id
+    eval $(openstack #{insecure} \
```
Comment (dirkmueller) [inline chef/cookbooks/magnum/recipes/setup.rb]: is anything actually installing openstack client? doesn't look like..

**Comment by dirkmueller** [comment]
I wonder if we shouldn't just do a "only_if { .. not already there .. }" clause instead of that complicated logic

File: chef/cookbooks/magnum/recipes/setup.rb (hunk 2 above)
Comment (mjura) [inline chef/cookbooks/magnum/recipes/setup.rb]: Adding new action is completely new feature, I agree with that but this is new card which has also influence on Heat for example.

File: chef/cookbooks/magnum/recipes/setup.rb (hunk 3 above)
Comment (mjura) [inline chef/cookbooks/magnum/recipes/setup.rb]: Adding new action is completely new feature, I agree with that but this is new card which has also influence on Heat for example.

File: chef/cookbooks/magnum/recipes/setup.rb (hunk 4 above)
Comment (mjura) [inline chef/cookbooks/magnum/recipes/setup.rb]: Thx, I will check it

Comment (mjura) [inline chef/cookbooks/magnum/recipes/setup.rb]: I've added `python-openstackclient` package to the list of packages

File: chef/cookbooks/magnum/recipes/setup.rb [hunk 5]
```diff
@@ -23,9 +23,11 @@
 trustee_domain_admin = node["magnum"]["trustee"]["domain_admin_name"]
 trustee_domain_admin_password = node["magnum"]["trustee"]["domain_admin_password"]
 
-# Install magnumclient package
+# Install openstack and magnu mclient packages
```
Comment (sayalilunkad) [inline chef/cookbooks/magnum/recipes/setup.rb]: s/magnu mclient/magnum client

**Comment by mjura** [comment]
I fixed logic and I did tests, works fine

File: chef/cookbooks/magnum/recipes/setup.rb (hunk 4 above)
Comment (dirkmueller) [inline chef/cookbooks/magnum/recipes/setup.rb]: thanks

**Comment by dirkmueller** [comment]
+1

**Comment by toabctl** [comment]
+1

**Pull request merged**